### DISCUSSION TITLE
Error: Hydration failed because the initial UI does not match what was rendered on the server.

### DISCUSSION THREAD

**Discussion by ghost** [wevm/wagmi#1012] [Q&A] [answered]
Hello!  
I am receiving the error (using NextJs)
`Error: Hydration failed because the initial UI does not match what was rendered on the server.`

Nothing seems to be fixing it, I searched through the issues and discussions, I did see a fix but the discussion was old and there was talk of having an inbuilt fix for it, are there any other ways to solve this problem or should I go ahead with the `onMount` solution?, thank you!

Link to code (if that's relevant): https://github.com/0xamey/connect-wallet.git

**Comment by elchanclasdeltianguis** [comment]
Hello, I have a similar problem:
when I use the useAccount hook to get isConnected and then use it like this
 {isConnected && <UserBalance />}
I get "Error: Text content does not match server-rendered HTML." hydration errors in nextjs.
The same happens if I try address or chain from useNetwork.
I guess its not fatal but it does throw errors also in production.
Are there solutions?
thanks!

**Comment by ghost** [comment]
I also encountered this problem

**Reply by 0xbulma** [reply]
Same here

**Comment by fdarian** [comment, marked as answer]
Avoid the use of `autoConnect` and set `persister: null`. It looks like SSR support is coming, but currently it's still an issue and there are some discussion regarding this #710 #542.
